### PR TITLE
fix: refuse non QName process ids

### PR DESCRIPTION
**Pull request by carmenberndt**
__Which issue does this PR address?__

Closes https://github.com/zeebe-io/zeebe-modeler/issues/237

### CODE REVIEW

**Comment by carmenberndt** [comment]
Hi @MaxTru, I had a look at the failing tests, but couldn't figure out why they are failing. Can you please help me out here?

**Comment by carmenberndt** [comment]
Never mind, I've found the issue :)

**Review by pinussilvestrus — COMMENTED** [review]
Thanks for your contribution! Two things before merging this one

1. can you think about test cases that verify that behavior?
2. can you squash your commits into a single one, fitting our guidelines?

```text
feat(call-activity-props): add validation to process id

Closes #237 
```

**Comment by pinussilvestrus** [comment]
@carmenberndt  any updates on this? Do you need help?

**Comment by carmenberndt** [comment]
Hi @pinussilvestrus! I've added a test case now :)

**Comment by carmenberndt** [comment]
Also squashed my commits now 👍

File: client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ import selectionModule from 'diagram-js/lib/features/selection';
 import modelingModule from 'bpmn-js/lib/features/modeling';
 import propertiesProviderModule from '..';
 import zeebeModdleExtensions from 'zeebe-bpmn-moddle/resources/zeebe';
+import { expect } from 'chai';
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js:39]: Can you remove this? It's not needed, we also include expect 👍 Besides that, it's ready to merge

Comment (carmenberndt) [reply client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js]: Sure, done 👍

File: client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js [hunk 2]
```diff
@@ -82,6 +82,39 @@ describe('customs - call activity', function() {
...
+        // when
+        triggerValue(input, '123', 'change');
+      }));
+
+      it('should fail', function() {
+
+        // then
+        const calledElement = getCalledElement(bo);
+
+        expect(calledElement).to.exist;
```
Comment (pinussilvestrus) [inline client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js:112]: What's an interesting question here: we check beforehand, [that the `zeebe:calledElement` does not exist](https://github.com/zeebe-io/zeebe-modeler/pull/258/files#diff-af3552ec25161c47622c461e32570772R95). If we enter an `invalid` we expect here to be created, although this first entered input value is not valid.

From my understanding, it should be expected that the `zeebe:calledElement` won't be created if the first input value is invalid. I could imagine that's related to the way the `ValidationAwareTextField` is working, so we couldn't do anything against it without internal changes. So I'd be fine with this.

What do you think @nikku @MaxTru @barmac ?

Comment (barmac) [reply client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js:112]: I'd say: don't create `zeebe:calledElement` if you can't assign `processId`. This way, there is no distinction between a freshly created Call Activity and a Call Activity where user tried to but failed to provide a `processId`.

File: client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js [hunk 3]
```diff
@@ -82,6 +82,39 @@ describe('customs - call activity', function() {
...
+        expect(getCalledElement(bo)).to.be.undefined;
+
+        const input = getInputField(
+          container,
+          'camunda-callActivity-processId',
+          'processId'
+        );
+
+        // when
+        triggerValue(input, '123', 'change');
```
Comment (barmac) [inline client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js:104]: Can we move it to an actual test case? This way we can re-use the `beforeEach` also for a successful test.

**Comment by nikku** [comment]
From what I see the only missing bit to get this PR ready to merge is https://github.com/zeebe-io/zeebe-modeler/pull/258#pullrequestreview-506336069.

Do you still plan to follow up on it @carmenberndt?

**Comment by carmenberndt** [comment]
Hi @nikku! Sorry for the long wait here, I'll do this today.

File: client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js (hunk 3 above)
Comment (carmenberndt) [reply client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js]: I don't really understand how this is supposed to work though, you can't pass an argument to the `beforeEach`. I moved a part of it to a separate function, not sure if that is what you wanted though.

Comment (barmac) [reply client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js]: That's exactly what I meant :). If you could rename the `init` function to something more descriptive, e.g. `changeInputValue`, I will be happy to merge the PR. So the final result would look like:

```javascript
it('should execute', function() {

  // when
  changeInputValue('validId');

  // then
  expect(...)
});
```

Comment (carmenberndt) [reply client/src/app/tabs/bpmn/custom/properties-provider/__tests__/CallActivitySpec.js]: Great :) Renamed it.

**Review by barmac — APPROVED** [review]
Thank you for your contribution :)

**Pull request merged**